src/library.js

```diff
@@ -2718,6 +2722,82 @@ const createPilingJs = (rootElement, initOptions = {}) => {
     );
   };
 
+  const allLabels = new Map();
+  let labelArray;
+
+  const setPileLabel = (pileState, pileId, reset = false) => {
+    const {
+      items,
+      itemLabel,
+      itemLabelColor,
+      itemLabelText,
+      piles
+    } = store.state;
+
+    if (!itemLabel) return;
+
+    const objective = expandLabelObjective(itemLabel);
+
+    if (!allLabels.size || reset) {
+      labelArray = Object.entries(piles).reduce((labels, [, pile]) => {
+        if (!pile.items.length) return labels;
+
+        const label = objective.flatMap(o => {
+          if (o.aggregator) {
+            return o.aggregator(
+              pile.items.map(itemId => o.property(items[itemId]))
+            );
+          }
+          return pile.items.map(itemId => o.property(items[itemId]));
+        });
+
+        return [...labels, ...label];
+      }, []);
+
+      labelArray = unique(labelArray);
+      labelArray.forEach((label, index) => {
+        allLabels.set(label, index);
+      });
+    }
+
+    const pileInstance = pileInstances.get(pileId);
+
+    if (!pileInstance) return;
+
+    const label = unique(
+      objective.flatMap(o => {
+        if (o.aggregator) {
+          return o.aggregator(
+            pileState.items.map(itemId => o.property(items[itemId]))
+          );
+        }
+        return pileState.items.map(itemId => o.property(items[itemId]));
+      })
+    );
+
+    const color = label.map(l =>
+      isFunction(itemLabelColor)
+        ? itemLabelColor(l, labelArray)
+        : itemLabelColor[allLabels.get(l)]
+    );
+
+    let text = [];
+
+    if (itemLabelText) {
+      text = label.map(l => {
```

Comment:
For performance reasons, we should create unique `PIXI.Text` elements here and reuse them on the piles. Text elements are just another form of textures, so we should be able to stick them into `PIXI.Sprite` and position the sprite.

Reply:
Hmm I don't understand how to reuse the `PIXI.Text` element... If we stick it to `PIXI.Sprite`, then shouldn't we create multiple sprites?

Reply:
Can you take a look here? https://github.com/flekschas/piling.js/pull/138/files#diff-bab0fc1bfe78f6cc1e691c24be5a357cR21-R32

The idea is this:
1. Create a new `pixiText = new PIXI.Text('some text')` element for the label
2. Render the label using `pixiText.updateText()`
3. Extract texture using `texture = pixiText.texture`
4. Pass the texture to the pile, which can create a sprite out of it

This approach is muuuch more performant than create a new text element every time. I can explain it in more detail in our next meeting :)